### ISSUE TITLE
TypeDefinitionCategory problem.

### ISSUE THREAD

**Issue by oleghailenko** [JanusGraph/janusgraph#1058]
Hello guys.
In [commint](https://github.com/JanusGraph/janusgraph/commit/f7ee75710ed7a1f22ce5ab657a9d4eaf2bd855a3) to enum `TypeDefinitionCategory` 3 elements was added. 
But, becouse they was added to the middle of the list, old databases is not able to work with new version of janusgraph.
Before this commit, enum element `INDEX_FIELD` had ordinal = 19, and serialized in storage backend as 19. But after that, element `UPDATE_CONNECTION_EDGE` became 19th element.

**Comment by twilmes** [comment]
@oleghailenko thanks for reporting this! Are you interested in submitting a PR? If not, we'll  get this addressed.

**Comment by oleghailenko** [comment]
Not sure about what kind of PR you are talking about. Not sure even is it could be fixed. If you move new enum item to the end of the list, db, created after initial commit will be broken. If you change way of emun serializing, all databases will be broken. So, the best way to do - do not make such changes in the future.

**Comment by twilmes** [comment]
You're correct, if somebody created a db using a SNAPSHOT build after this commit, it will be broken but 0.3.0 hasn't been released yet so this commit should not be in production yet for anyone unless I'm misunderstanding your point.
